### PR TITLE
Add network and persistence configuration to firestore

### PR DESCRIPTION
**Pull request by arthuro555**
Adds options enable/disable [persistence](https://firebase.google.com/docs/reference/js/firebase.firestore.Firestore#enablepersistence) and [network](https://firebase.google.com/docs/reference/js/firebase.firestore.Firestore#enablenetwork) on Firestore.

### CODE REVIEW

**Comment by arthuro555** [comment]
This PR is ready for review.

File: Extensions/Firebase/JsExtension.js [hunk 1]
```diff
@@ -1048,6 +1048,102 @@ module.exports = {
...
+      )
+      .setFunctionName('firebase.firestore().enablePersistence');
+
+    extension
+      .addAction(
+        'FirestoreDisablePersistence',
+        _('Disable persistence'),
+        _(
+          'Disables the storing of fetched data and clear all the data that has been stored.\n' +
+            'This needs to be called before any other firestore operation, else it will fail.'
```
Comment (4ian) [inline Extensions/Firebase/JsExtension.js:1083]: otherwise it will fail

File: Extensions/Firebase/JsExtension.js [hunk 2]
```diff
@@ -1048,6 +1048,102 @@ module.exports = {
...
 
+    extension
+      .addAction(
+        'FirestoreEnablePersistence',
+        _('Enable persistence'),
+        _(
+          'When persistence is enabled, all data that is fetched ' +
+            'from the database is being automatically stored to allow to ' +
+            'continue accessing the data if cut off from the network, instead of waiting for reconnection.\n' +
+            'This needs to be called before any other firestore operation, else it will fail.'
```
Comment (4ian) [inline Extensions/Firebase/JsExtension.js:1059]: otherwise it will fail

File: Extensions/Firebase/JsExtension.js [hunk 3]
```diff
@@ -1048,6 +1048,102 @@ module.exports = {
...
+        'Extensions/Firebase/B_firebasetools/D_cloudfirestoretools.js'
+      )
+      .setFunctionName('firebase.firestore().clearPersistence');
+
+    extension
+      .addAction(
+        'FirestoreEnableNetwork',
+        _('Reenable network'),
+        _('Reenables the connection to the database after disabling it.'),
+        _('Reenable network'),
```
Comment (4ian) [inline Extensions/Firebase/JsExtension.js:1106]: ubernit: Re-enable :)

**Review by 4ian — APPROVED** [review]
Looks great :)

**Pull request merged**